pkg/controller/release/release_controller.go

```diff
@@ -350,11 +350,16 @@ func (c *Controller) syncOneReleaseHandler(key string) error {
 			err.Error(),
 		)
 		releaseutil.SetReleaseCondition(&rel.Status, *condition)
+	}
 
-		if _, updateErr := c.clientset.ShipperV1alpha1().Releases(namespace).Update(rel); updateErr != nil {
-			return shippererrors.NewKubeclientUpdateError(rel, updateErr)
+	// any better ideas?
+	if !reflect.DeepEqual(curRel, rel) {
```

Comment:
If `rel` is created in the block above, then `chooseClusters` could be used to detect whether or not the update should happen.

Reply:
It's a but tricky here: there is only 1 Update statement and a few code ways that can modify the rel object:
* ChooseClusters chooses relevant placement and puts this data into annotations
* scheduling succeeds and sets Scheduled condition to true
* scheduling fails and sets Scheduled condition to false
All 3 paths lead to a single Update point which triggers if there is effectively any difference between rel and curRel. From this perspective there are 2 components (effectively 3) that should indicate whether a change has been made.